components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -1246,144 +1434,167 @@ private void deleteQueryParamCondition(Connection connection, int conID) throws
 
-        try (Connection connection = DAOUtil.getConnection();
-                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery))    {
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery))    {
             preparedStatement.setString(1, uuid);
             preparedStatement.execute();
-        } catch (SQLException e)    {
-            String msg = "An Error occurred while deleting Policy with uuid [" + uuid + "], ";
-            log.error(msg, e);
-            throw new APIMgtDAOException(msg, e);
         }
     }
 
-    private static void updateAPIPolicy(String uuid, Connection connection, String name, String displayName,
-            String description, String quotaType, int quota, long unitTime, String timeUnit, String applicableLevel)
-            throws SQLException {
-        final String query = "UPDATE AM_API_POLICY SET NAME = ?, DISPLAY_NAME = ?, DESCRIPTION = ?, "
-                + "DEFAULT_QUOTA_TYPE = ?, DEFAULT_QUOTA = ?, DEFAULT_UNIT_TIME = ?, DEFAULT_TIME_UNIT = ?, "
-                + "APPLICABLE_LEVEL = ? WHERE UUID = ?";
+    /**
+     * Updates an existing API policy
+     *
+     * @param policy {@link APIPolicy} instance
+     * @param connection DB Connection instance
+     * @throws SQLException if an error occurs while updating the API policy
+     * @throws APIMgtDAOException if the uuid of the policy is not available
+     */
+    private void updateAPIPolicy(APIPolicy policy, Connection connection) throws SQLException, APIMgtDAOException {
 
-        try (PreparedStatement statement = connection.prepareStatement(query)) {
-            statement.setString(1, name);
-            statement.setString(2, displayName);
-            statement.setString(3, description);
-            statement.setString(4, quotaType);
-            statement.setInt(5, quota);
-            statement.setLong(6, unitTime);
-            statement.setString(7, timeUnit);
-            statement.setString(8, applicableLevel);
-            statement.setString(9, uuid);
+        String queryFindPolicyName = "SELECT * from AM_API_POLICY WHERE UUID = ?";
 
-            statement.execute();
+        if (policy.getUuid() == null || policy.getUuid().isEmpty()) {
+            String errorMsg = "Policy uuid is not found, unable to update policy: " + policy.getPolicyName();
+            throw new APIMgtDAOException(errorMsg);
+        }
+
+        try (PreparedStatement selectStatement = connection.prepareStatement(queryFindPolicyName)) {
+            selectStatement.setString(1, policy.getUuid());
+            try (ResultSet resultSet = selectStatement.executeQuery()) {
+                while (resultSet.next()) {
+                    String policyName = resultSet.getString(APIMgtConstants.ThrottlePolicyConstants.
+                            COLUMN_NAME);
+                    deletePolicy(policyName, APIMgtConstants.ThrottlePolicyConstants.API_LEVEL);
```

Comment:
Any reason why we cannot simply do an update in this case, instead of doing a delete and add?